File: frontend/admin/src/app/models/ruta.ts
export interface Ruta {
  $key?: any;
  origin: Location;
  name: string;
  destination: Location;
  waypoints?: Location[];
}
export class Location {
  lat: number;
  lng: number;
  location?: any;
}
export class WayPoint {
  location: { location: Location };
}

File: frontend/usuario_app/lib/ruta.dart
import 'dart:ffi';

import 'package:firebase_database/firebase_database.dart';

class Ruta {
  // ignore: unused_field
  DatabaseReference _id;
  String name;
  Location origin;
  Location destination;
  List<Waypoint> waypoints;
}

class Location {
  Float lat;
  Float lng;
}

class Waypoint {
  Location location;
  bool stopover;
}
